### PR TITLE
Set up webpack for web and node

### LINKED ISSUE #40: Reevaluate Webpack setup
As per https://github.com/OpenSTFoundation/mosaic.js/pull/39#pullrequestreview-198168195:

> Is there a ticket/plan to get rid of the webpack stuff?

If I interpret it correctly, webpack is currently mostly used to produce seperate builds for browser and node (excluding the `//__NOT_FOR_WEB__BEGIN__` blocks). This was required because the contract ABIs and bins were read from disk via the `fs` module, which isn't available in the browser. This requirement should be resolved with #39, clearing the way for a unified build.

There might however be other reasons for why the webpack build is neccessary (we probably still need it for minifing, and might also want to have it around if/when we adopt Flow/Typescript).

### PR DESCRIPTION
**Pull request by hobofan**
- Got rid of the `//__NOT_FOR_WEB__BEGIN__` blocks and the script for that
- Webpack build now builds seperate bundles for web(=browser) and node, and added `package.json` keys accordingly
- Added `npm run build` to Travis to make sure build stays intact

To allow for per-target requires (which are required so that node system libraries are not required during web build), webpack resolves `AbiBinProvider-node.js` module to `AbiBinProvider-node.js` or `AbiBinProvider-web.js` depending on the build target (`-node` is the default instead of a placeholder, so that the module correctly resolves to node in our dev environment without a build step).

FIXES #40

### CODE REVIEW

File: src/AbiBinProvider.js [hunk 1]
```diff
@@ -47,19 +24,19 @@ class AbiBinProvider {
 
     let abi;
     if (typeof abiFileContent === 'string') {
-      //Parse it.
+      // Parse it.
```
Comment (schemar) [inline src/AbiBinProvider.js]: Redundant

File: src/AbiBinProvider.js [hunk 2]
```diff
@@ -74,15 +51,15 @@ class AbiBinProvider {
     oThis.custom = oThis.custom || {};
 
     if (typeof binFileContent !== 'string') {
-      //Parse it.
-      let err = new Error('Bin should be a string');
+      // Parse it.
```
Comment (schemar) [inline src/AbiBinProvider.js]: redundant

File: src/AbiBinProvider-node.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
+'use strict';
```
Comment (schemar) [inline src/AbiBinProvider-node.js:21]: license

File: src/AbiBinProvider-web.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+'use strict';
```
Comment (schemar) [inline src/AbiBinProvider-web.js:21]: license

File: src/AbiBinProvider-web.js [hunk 5]
```diff
@@ -0,0 +1,5 @@
...
+
+module.exports = {
+  loadContracts: () => {},
```
Comment (schemar) [inline src/AbiBinProvider-web.js]: shouldn't we throw so that the consumer knows it cannot be used?

File: src/AbiBinProvider-node.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    );
+    throw err;
+  }
+
+  // add all ABIs from abiFolderPath
+  fs.readdirSync(abiFolderPath).forEach((abiFile) => {
+    const contractName = path.basename(abiFile, path.extname(abiFile));
+    const contractAbi = JSON.parse(fs.readFileSync(abiFile));
+    provider.addABI(contractName, contractAbi);
+  });
```
Comment (schemar) [inline src/AbiBinProvider-node.js:56]: don't we use the `mosaic-contracts` npm package?

File: src/AbiBinProvider.js [hunk 7]
```diff
@@ -2,41 +2,18 @@
```
Comment (schemar) [inline src/AbiBinProvider.js:28]: license

File: src/AbiBinProvider.js [hunk 8]
```diff
@@ -2,41 +2,18 @@
...
-      );
-      throw err;
-    }
-    if (!path.isAbsolute(binFolderPath)) {
-      let err = new Error(
-        '"binFolderPath" is not Absolute. Please provide absolute path.',
-      );
-      throw err;
+
+    // only does something on nodejs
```
Comment (schemar) [inline src/AbiBinProvider.js]:
```suggestion
    // loadContracts() returns an empty object in the browser.
```

File: tests/utils/MockContractsDeployer.js [hunk 9]
```diff
@@ -1,9 +1,7 @@
 'use strict';
 
-//__NOT_FOR_WEB__BEGIN__
-const fs = require('fs'),
-  path = require('path');
-//__NOT_FOR_WEB__END__
+const fs = require('fs');
```
Comment (schemar) [inline tests/utils/MockContractsDeployer.js:23]: license

**Review by schemar — COMMENTED** [review]
Good! Just some comments. Where is the replacement of the import done?

File: src/AbiBinProvider.js (hunk 8 above)
Comment (hobofan) [reply src/AbiBinProvider.js]: It doesn't return an empty object, it is a noop.

File: src/AbiBinProvider-node.js (hunk 6 above)
Comment (hobofan) [reply src/AbiBinProvider-node.js:56]: Yes, but the consumers in brandedtoken.js and openst.js still assume that the contracts can be loaded from file system via that interface. I'm not sure if we should add a deprecation notice for that already?

File: src/AbiBinProvider-web.js (hunk 5 above)
Comment (hobofan) [reply src/AbiBinProvider-web.js]: The idea is that it is used on either platform but it just falls back to a noop when it is not supported. The only direct consumer here is also `AbiBinProvider`, as it is not exposed in the package.

File: src/AbiBinProvider.js (hunk 8 above)
Comment (schemar) [reply src/AbiBinProvider.js]: Ha! Still learning to read JS 😅 
The comment should still be an English sentence with a period.

**Comment by hobofan** [comment]
@schemar The module replacement: https://github.com/OpenSTFoundation/mosaic.js/blob/d0fde5d8a6dc0f3720bfa7e5598c2d5dfa48ad09/webpack.config.js#L22

**Review by schemar — APPROVED** [review]
☑️
Just a few smaller comments remaining.

**Comment by schemar** [comment]
👍:shipit:

**Pull request merged**